lib/src/lints/avoid_unused_parameters/avoid_unused_parameters_rule.dart

```diff
@@ -91,12 +94,16 @@ class AvoidUnusedParametersRule extends SolidLintRule {
     ErrorReporter reporter,
     CustomLintContext context,
   ) {
-    context.registry.addCompilationUnit((node) {
-      final visitor = AvoidUnusedParametersVisitor();
-      node.accept(visitor);
+    context.registry.addDeclaration((node) {
+      final isIgnored = config.parameters.exclude.shouldIgnore(node);
+      
+      if (!isIgnored) {
```

Comment:
[Prefer early returns](https://lints.solid.software/docs/custom_lints/prefer_early_return)
```suggestion
      if (isIgnored) return;
```